osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -21,12 +21,12 @@ protected override void OnSelectionChanged()
         {
             base.OnSelectionChanged();
 
-            bool canOperate = EditorBeatmap.SelectedHitObjects.Count > 1 || EditorBeatmap.SelectedHitObjects.Any(s => s is Slider);
+            Quad quad = selectedMovableObjects.Length > 0 ? getSurroundingQuad(selectedMovableObjects) : new Quad();
 
-            SelectionBox.CanRotate = canOperate;
-            SelectionBox.CanScaleX = canOperate;
-            SelectionBox.CanScaleY = canOperate;
-            SelectionBox.CanReverse = canOperate;
+            SelectionBox.CanRotate = quad.Width > 0 || quad.Height > 0;
+            SelectionBox.CanScaleX = quad.Width > 0;
+            SelectionBox.CanScaleY = quad.Height > 0;
+            SelectionBox.CanReverse = EditorBeatmap.SelectedHitObjects.Count > 1 || EditorBeatmap.SelectedHitObjects.Any(s => s is Slider);
```

Comment:
Although on the other side I'd argue this should also probably be disabled for stacks.

Reply:
I did it this way because this also handles spinners, which are not movable but can be reversed in time. I just noticed that that doesn't work in lazer but it does in stable.

Reply:
I guess you can *theoretically* map a perfect stack that isn't temporally symmetrical, something like

![2020-11-12-235800_717x806_scrot](https://user-images.githubusercontent.com/20418176/99006546-ec739880-2542-11eb-90e2-74d9e1367bc7.png)

but it looks super off and also not sure if that's a thing that's even valid/reasonable mapping...

Reply:
This doesn't only apply to perfect stacks, stacks with offset as well.
Also say it is valid, [dum surfer](https://osu.ppy.sh/beatmapsets/716193#osu/1513180) is an example that comes to mind, and probably a lot of old maps with stacks.
![image](https://user-images.githubusercontent.com/15140758/99006948-8a676300-2543-11eb-8323-03e0b6b6fa80.png)